src/apns_connection.erl

```diff
@@ -336,12 +346,13 @@ down(internal
        , client          := Client
        , backoff         := Backoff
        , backoff_ceiling := Ceiling
-       }) ->
+       } = StateData0) ->
   true = demonitor(GunMon, [flush]),
   gun:close(GunPid),
   Client ! {reconnecting, self()},
-  Sleep = backoff(Backoff, Ceiling) * 1000,
-  {keep_state_and_data, {state_timeout, Sleep, backoff}};
+  StateData = StateData0#{ backoff => backoff(Backoff, Ceiling) },
+  Sleep = maps:get(backoff, StateData) * 1000,
```

Comment:
Why fetch backoff again from map instead of using already computed value, for example:

```erl
Backoff = backoff(Backoff0, Ceiling),
StateData = StateData0#{backoff => Backoff},
Sleep = Backoff * 1000
```

Reply:
fixed